docs/mkdocs/docs/api/macros/json_diagnostic_positions.md

```diff
@@ -50,18 +49,25 @@ When the macro is not defined, the library will define it to its default value.
 
 !!! note "CMake option"
 
-    Diagnostic messages can also be controlled with the CMake option
+    Diagnostic positions can also be controlled with the CMake option
     [`JSON_Diagnostic_Positions`](../../integration/cmake.md#json_diagnostic_positions) (`OFF` by default)
     which defines `JSON_DIAGNOSTIC_POSITIONS` accordingly.
 
+!!! note "Availability"
+
+    Diagnostic positons are only available if the value was created by the [`parse`](../basic_json/parse.md) function.
+    The [`sax_parse`](../basic_json/sax_parse.md) function or all other means to create a JSON value **do not** set the
```

Comment:
I thought I saw them being set through `sax_parse` as well.

Reply:
No, `sax_parse` is never creating `basic_json` objects. Only in the combination of `sax_parse` using the `json_sax_dom_callback_parser`, the diagnostic positions are set. This only works, because `json_sax_dom_callback_parser` is a friend. This is the code path that the `parse` function is using.

Any other use of the `sax_parse` function does not allow creating `basic_json` values and setting the private position variables.

Possible fix: make these variables public or add a setter?

Reply:
Okay, that's what it was, thanks.